### PR TITLE
Make it so StandardD2sV3 can only be used in test

### LINKED ISSUE #707: disable ability to use StandardD2sV3 except when running in test
As a quality engineer I want to be able to run OSA using a tiny VM for test purposes or CI.
As an end user I don't want to be able to run OSA using a tiny VM because it almost certainly won't be powerful enough for me and I'll raise a support ticket because it breaks.

I think we should refuse to validate use of StandardD2sV3 except when running in test, but that implies making our validation code dependent on configuration, which might open a can of works.  @kargakis @pweil- what's your take?

### PR DESCRIPTION
**Pull request by thekad**
Fixes #707 

This should error out when requesting StandardD2sV3 in production
environments, as that size is only really useful for QA/testing

### CODE REVIEW

File: pkg/api/validate.go [hunk 1]
```diff
@@ -334,7 +344,11 @@ func validateAgentPoolProfiles(apps []AgentPoolProfile, vnet *net.IPNet) (errs [
 	return
 }
 
-func validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
+func (v *Validator) validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
+	if !v.PluginConfig.TestConfig.RunningUnderTest && isTestVMSize(app.VMSize) {
+		errs = append(errs, fmt.Errorf("invalid VM size %s", app.VMSize))
+	}
+
```
Comment (asalkeld) [inline pkg/api/validate.go]: wouldn't it be better to turn validMasterAndInfraVMSizes (and friends) into functions that takes the config and returns the correct data? @jim-minter (would we ever have different sizes in different regions or dependent on other options?)

File: pkg/api/validate_test.go [hunk 2]
```diff
@@ -362,6 +362,13 @@ func TestValidate(t *testing.T) {
 		},
 	}
 
+	tc := TestConfig{
+		RunningUnderTest: true,
+	}
```
[the remark this answers is not in the document]
Comment (thekad) [reply pkg/api/validate_test.go]: @asalkeld see fb617b4f7 and lmk if that's enough

**Comment by kargakis** [comment]
From the [test artifacts](https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_openshift-azure/746/pull-ci-openshift-openshift-azure-master-e2e-azure/894/artifacts/e2e-azure/), it seems there is an issue with the sync pod
```
NAMESPACE     NAME                        READY     STATUS    RESTARTS   AGE       IP         NODE
kube-system   controllers-master-000000   1/1       Running   0          22m       10.0.0.4   master-000000
kube-system   controllers-master-000001   1/1       Running   0          22m       10.0.0.5   master-000001
kube-system   controllers-master-000002   1/1       Running   0          22m       10.0.0.6   master-000002
kube-system   master-api-master-000000    1/1       Running   0          21m       10.0.0.4   master-000000
kube-system   master-api-master-000001    1/1       Running   0          22m       10.0.0.5   master-000001
kube-system   master-api-master-000002    1/1       Running   1          22m       10.0.0.6   master-000002
kube-system   master-etcd-master-000000   1/1       Running   0          21m       10.0.0.4   master-000000
kube-system   master-etcd-master-000001   1/1       Running   0          22m       10.0.0.5   master-000001
kube-system   master-etcd-master-000002   1/1       Running   0          22m       10.0.0.6   master-000002
kube-system   sync-master-000000          0/1       Error     9          22m       10.0.0.4   master-000000
```

File: pkg/api/validate.go [hunk 3]
```diff
@@ -334,7 +344,11 @@ func validateAgentPoolProfiles(apps []AgentPoolProfile, vnet *net.IPNet) (errs [
 	return
 }
 
-func validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
+func (v *Validator) validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
+	if !v.PluginConfig.TestConfig.RunningUnderTest && isTestVMSize(app.VMSize) {
+		errs = append(errs, fmt.Errorf("invalid VM size %s", app.VMSize))
```
Comment (jim-minter) [inline pkg/api/validate.go]: error message doesn't align with equivalent error messages below

File: pkg/api/validate.go (hunk 1 above)
Comment (jim-minter) [reply pkg/api/validate.go]: agreed.

**Comment by jim-minter** [comment]
force pushed with review fixes

**Pull request merged**